packages/core/src/browser/messaging/ws-connection-provider.ts

```diff
@@ -143,3 +129,20 @@ export class WebSocketConnectionProvider extends AbstractConnectionProvider<WebS
         this.onSocketDidCloseEmitter.fire(undefined);
     }
 }
+
+function toIWebSocket(socket: Socket): IWebSocket {
```

Comment:
Since we are using this non-public function in a protected/overridable context, I would be in favor of making this either exported or part of the class and `protected`.

Reply:
Good point. I refactored the `toIWebsocket` functions to be part of the class.